### ISSUE TITLE
Add notification when LSP/ccls is ready

### ISSUE THREAD

**Issue by JohnC32** [MaskRay/emacs-ccls#60]
The current workflow for a large C++ project I'm setting up using lsp-mode with ccls is to:

1. Pregenerate the index using ccls -init="..." on our project.This takes many hours using 12 threads because of the size - many millions of lines of code.
2. Launch emacs and when a C++ file is loaded, activate lsp-mode.
The activation of lsp mode does take a while. Looking at "ccls::stderr" I see it takes a couple of minutes from the first line to the following "ready" line:

    10:19:10 ccls             pipeline.cc:485 I loaded project. Refresh semantic highlight for all working file.

Specifically, using lsp-mode, https://github.com/emacs-lsp/lsp-mode/commit/66780e026da6711a4219802bc20b7cb165ea44b2, with ccls, I immediately see the following messages:

    LSP :: Connected to [ccls:64240 status:starting].
    LSP :: ccls:64240 initialized successfully
    LSP :: not indexed [2 times]

I then switch to the ccls::stderr buffer and see many thousands of "I loaded cache for file.cpp", then a couple minutes later in the ccls::stderr buffer, I see that lsp/ccls is ready when it prints the line:

    12:11:10 ccls             pipeline.cc:485 I loaded project. Refresh semantic highlight for all working file.

Would it be possible to add a notification when lsp/ccls is ready, e.g. (message "LSP ready - reload files to refresh semantic highlighting")? Even better - it would be nice if the C++ buffers refreshed, so I don't need to manually refresh.

See https://github.com/emacs-lsp/lsp-mode/issues/1114 in lsp-mode where I originally requested this, but later found the request belonged here.

Thanks

**Comment by MaskRay** [comment]
For a large project, it may take a while to load all the index files. ccls can load the index files of llvm+clang+lld+compiler-rt+libcxx (1G) in 1 minute on my laptop.

You may read https://github.com/MaskRay/ccls/issues/485#issuecomment-533039040 about the initialization option `request.timeout`

**Comment by JohnC32** [comment]
I'm working with a project much larger than llvm, etc. On a fairly high-end desktop computer I see it taking 2 to 3 minutes for ccls to get to the "I loaded project" state. It would be nice if I got a notification when CCLS was done loading. Could an async activity monitor the stderr buffer and when the "I loaded project" is complete, a ready message gets displayed?

The request.timeout seems to be more geared toward handling large files. This is helpful because we have many that are extremely large, but I think this is different from this ccls startup / ready notification message request.

As an aside, I'm switching from RTags to CCLS and finding CCLS better. RTags does stall too much, though I never full got to the bottom of the stall. CCLS is very responsive and works well. The only downside of CCLS is that it uses 10X the amount of memory of RTags and much more disk space for the index.
